Week3/homework/step4-bonus.js

```diff
@@ -4,7 +4,15 @@ const values = ['a', 'b', 'c', 'd', 'a', 'e', 'f', 'c'];
 
 // Add your function here. Try and come up with a good name for this function
 
+function constructArrayWithUniqueValues(arr) {
+  const newValues = [];
+  for (const element of arr) {
+    if (newValues.includes(element) === false) {
+      newValues.push(element);
+    }
+  }
+  return newValues;
+}
 // Replace `yourFunction` with the name of the function you just created
-const uniqueValues = yourFunction(values);
-
+const uniqueValues = constructArrayWithUniqueValues(values);
 console.log(uniqueValues);
```

Comment:
Really well done the code! Especially I appreciate you payed attention to naming the function and the variables also! 👍